src/poetry/inspection/info.py

```diff
@@ -490,7 +490,10 @@ def from_directory(cls, path: Path, disable_build: bool = False) -> PackageInfo:
         else:
             info = cls.from_metadata_directory(path)
 
-            if not info or info.requires_dist is None:
+            # unfortunately, we cannot yet trust that an empty or None requires_dist
```

Comment:
This

Reply:
I agree. In https://github.com/python-poetry/poetry/pull/9019#discussion_r1501861491 and the following we concluded that `None` means "I don't know" and `[]` means we are sure that there are no dependencies so this change really feels like a step backwards.

Reply:
Let me check the test cases again, but disabling net access ended up using the mocked http calls. Which led to this code path being triggered causing metadata to be not correctly discovered since the metadata did not contain a requires-dist. 

But, either way, let me see if I can revert this.

Reply:
Okay, seems the problem fixed itself with the refreshed test data. So, dropping the commit for now.